test/e2e/auth/pod_security_policy.go

```diff
@@ -0,0 +1,333 @@
+		})
+	})
+})
+
+func expectForbidden(err error) {
+	Expect(err).To(HaveOccurred(), "should be forbidden")
+	Expect(apierrs.IsForbidden(err)).To(BeTrue(), "should be forbidden error")
+}
+
+func testPrivilegedPods(f *framework.Framework, tester func(pod *v1.Pod)) {
+	By("Running a privileged pod", func() {
+		privileged := restrictedPod(f, "privileged")
+		privileged.Spec.Containers[0].SecurityContext.Privileged = boolPtr(true)
+		privileged.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = nil
+		tester(privileged)
+	})
+
+	By("Running a HostPath pod", func() {
+		hostpath := restrictedPod(f, "hostpath")
+		hostpath.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{{
+			Name:      "hp",
+			MountPath: "/hp",
+		}}
+		hostpath.Spec.Volumes = []v1.Volume{{
+			Name: "hp",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{Path: "/tmp"},
+			},
+		}}
+		tester(hostpath)
+	})
+
+	By("Running a HostNetwork pod", func() {
+		hostnet := restrictedPod(f, "hostnet")
+		hostnet.Spec.HostNetwork = true
+		tester(hostnet)
+	})
+
+	By("Running a HostPID pod", func() {
+		hostpid := restrictedPod(f, "hostpid")
+		hostpid.Spec.HostPID = true
+		tester(hostpid)
+	})
+
+	By("Running a HostIPC pod", func() {
+		hostipc := restrictedPod(f, "hostipc")
+		hostipc.Spec.HostIPC = true
+		tester(hostipc)
+	})
+
+	if common.IsAppArmorSupported() {
+		By("Running a custom AppArmor profile pod", func() {
+			aa := restrictedPod(f, "apparmor")
+			// Every node is expected to have the docker-default profile.
+			aa.Annotations[apparmor.ContainerAnnotationKeyPrefix+"pause"] = "localhost/docker-default"
+			tester(aa)
+		})
+	}
+
+	By("Running an unconfined Seccomp pod", func() {
+		unconfined := restrictedPod(f, "seccomp")
+		unconfined.Annotations[v1.SeccompPodAnnotationKey] = "unconfined"
+		tester(unconfined)
+	})
+
+	By("Running a CAP_SYS_ADMIN pod", func() {
+		sysadmin := restrictedPod(f, "sysadmin")
+		sysadmin.Spec.Containers[0].SecurityContext.Capabilities = &v1.Capabilities{
+			Add: []v1.Capability{"CAP_SYS_ADMIN"},
+		}
+		sysadmin.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = nil
+		tester(sysadmin)
+	})
+}
+
+func createAndBindPSP(f *framework.Framework, pspTemplate string) (cleanup func()) {
+	// Create the PodSecurityPolicy object.
+	json, err := utilyaml.ToJSON([]byte(pspTemplate))
+	framework.ExpectNoError(err)
+	psp := &extensionsv1beta1.PodSecurityPolicy{}
+	framework.ExpectNoError(runtime.DecodeInto(legacyscheme.Codecs.UniversalDecoder(), json, psp))
+	// Add the namespace to the name to ensure uniqueness and tie it to the namespace.
+	ns := f.Namespace.Name
+	name := fmt.Sprintf("%s-%s", ns, psp.Name)
+	psp.Name = name
+	_, err = f.ClientSet.ExtensionsV1beta1().PodSecurityPolicies().Create(psp)
+	framework.ExpectNoError(err, "Failed to create PSP")
+
+	// Create the Role to bind it to the namespace.
+	_, err = f.ClientSet.RbacV1beta1().Roles(ns).Create(&rbacv1beta1.Role{
```

Comment:
we have helper framework methods for binding clusterroles (BindClusterRole/BindClusterRoleInNamespace)... wonder if these should be made helpers as well

notably, those tolerate failure, since RBAC is not required to be enabled for e2e... if you're requiring it for this, we might want to fail earlier